client/my-sites/sharing/connections/connection.jsx

```diff
@@ -1,159 +1,175 @@
+import classNames from 'classnames';
+import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-var analytics = require( 'lib/analytics' ),
-	serviceConnections = require( './service-connections' );
-
-module.exports = React.createClass( {
-	displayName: 'SharingConnection',
-
-	propTypes: {
-		site: React.PropTypes.object,                    // The site for which the connection was created
-		user: React.PropTypes.object,                    // A user object
-		connection: React.PropTypes.object.isRequired,   // The single connection object
-		service: React.PropTypes.object.isRequired,      // The service to which the connection is made
-		onDisconnect: React.PropTypes.func,              // Handler to invoke when disconnecting
-		isDisconnecting: React.PropTypes.bool,           // Is a service disconnection request pending?
-		showDisconnect: React.PropTypes.bool,            // Display an inline disconnect button
-		onRefresh: React.PropTypes.func,                 // Handler to invoke when refreshing
-		isRefreshing: React.PropTypes.bool,              // Is a service refresh request pending?
-		onToggleSitewideConnection: React.PropTypes.func // Handler to invoke when toggling sitewide connection
-	},
-
-	getInitialState: function() {
-		return { isSavingSitewide: false };
-	},
-
-	componentDidUpdate: function( prevProps ) {
+import { getCurrentUser } from 'state/current-user/selectors';
+import { getSelectedSite } from 'state/ui/selectors';
+import { getUser } from 'state/users/selectors';
+import { recordGoogleEvent } from 'state/analytics/actions';
+
+class SharingConnection extends Component {
+	static propTypes = {
+		site: PropTypes.object,                    // The site for which the connection was created
+		user: PropTypes.object,                    // A user object
+		connection: PropTypes.object.isRequired,   // The single connection object
+		service: PropTypes.object.isRequired,      // The service to which the connection is made
+		onDisconnect: PropTypes.func,              // Handler to invoke when disconnecting
+		isDisconnecting: PropTypes.bool,           // Is a service disconnection request pending?
+		showDisconnect: PropTypes.bool,            // Display an inline disconnect button
+		onRefresh: PropTypes.func,                 // Handler to invoke when refreshing
+		isRefreshing: PropTypes.bool,              // Is a service refresh request pending?
+		onToggleSitewideConnection: PropTypes.func // Handler to invoke when toggling sitewide connection
+	};
+
+	static defaultProps = {
+		isDisconnecting: false,
+		isRefreshing: false,
+		onDisconnect: () => {},
+		onRefresh: () => {},
+		onToggleSitewideConnection: () => {},
+		showDisconnect: false,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			isSavingSitewide: false
+		};
+
+		this.toggleSitewideConnection = this.toggleSitewideConnection.bind( this );
+	}
+
+	componentDidUpdate( prevProps ) {
```

Comment:
Aside (existed previously): It would be nice if this were `componentWillReceiveProps` to save an unnecessary additional render from the `this.setState`.

https://facebook.github.io/react/docs/react-component.html#componentwillreceiveprops